data/src/main/java/com/microsoft/azure/kusto/data/http/HttpClientFactory.java

```diff
@@ -23,69 +17,52 @@ public class HttpClientFactory {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientFactory.class);
 
+    // Todo: for team consideration...
+    // With previous versions, this was configurable per request, now it's only configurable per client
+    // Because there is only 1 HTTP Client per KustoClient (ClientImpl) abstraction, this can only be configured universally for the instance
+    // For now I've put this value here as a placeholder, however this should be added to ClientOptions
+    private static final Long TIMEOUT = Duration.ofMinutes(10).toMillis();
```

Comment:
We can actually still create a client per client, using the Builder instead of the Provider, and I think we may want that?

Reply:
Moved this into the builder per request. [See this commit.](https://github.com/Azure/azure-kusto-java/pull/335/commits/abae5fb150f99e7e8ab5925a9b81fc73ad80ad78)

Reply:
Added client provider to the builder while I was in there as well, although if you are open to the idea, I would like to discuss deprecating the HttpClientProperties class in the future. [See this comment.](https://github.com/Azure/azure-kusto-java/pull/335#issuecomment-1944673160)